File: tests/fixtures/js_type_infer/square.ts
// Homonyme volontaire : Square.area existe pour prouver que le fan-out par nom
// (comportement AVANT inference) creerait une arete parasite render->square.ts::area,
// et que l'inference de type l'elimine (const c = new Circle() => c.area() ne vise que circle.ts::area).
export class Square {
  area() {
    return 4.0;
  }
}

File: tests/fixtures/py_type_infer/square.py
# Homonyme volontaire : Square.area existe pour prouver que le fan-out par nom
# (comportement AVANT inference) cree une arete parasite main->square.py::area,
# et que l'inference de type l'elimine (c = Circle() => c.area() ne vise que circle.py::area).
class Square:
    def area(self):
        return 4.0
